pilot/pkg/model/jwks_resolver.go

```diff
@@ -197,9 +210,14 @@ func (r *jwksResolver) GetPublicKey(jwksURI string) (string, error) {
 	}
 
 	// Fetch key if it's not cached, or cached item is expired.
+	// The cache expiration happens if the jwt public key refresh have been constantly failing for
```

Comment:
If the key is in the cache, it will be fetched from cache all the time until expired, right? In other words, you reach here not because of key refresh have been constantly failing, but simply because this code path (and the followed) `getRemoteConttent` hadn't been called for a week.

Reply:
No, the refresh job fetch the key independently from the expire time after this PR, the refresh job will be triggered every 20 minutes, and will first try to evict and update all the public key in the cache.

We reach here because the refresh job failed to update the key from network for a whole 1 week, because if it did, it will update the expire time to another 1 week.

I need to update the comments here.